libs/payments/stripe/src/lib/subscription-item.factory.ts

```diff
@@ -0,0 +1,85 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { faker } from '@faker-js/faker';
+import { Stripe } from 'stripe';
+
+export const SubscriptionItemFactory = (
+  subscriptionId: string,
+  productId: string,
+  planId: string,
+  override: Partial<Stripe.SubscriptionItem>
+): Stripe.SubscriptionItem => {
+  return {
+    id: `si_${faker.string.alphanumeric({ length: 14 })}`,
+    object: 'subscription_item',
+    billing_thresholds: null,
+    created: 1696897253,
+    metadata: {},
+    plan: {
+      id: planId,
+      object: 'plan',
+      active: true,
+      aggregate_usage: null,
+      amount: 500,
+      amount_decimal: '500',
+      billing_scheme: 'per_unit',
+      created: 1583259953,
+      currency: 'usd',
+      interval: 'month',
+      interval_count: 1,
+      livemode: false,
+      metadata: {
+        productOrder: '1',
+      },
+      nickname: '123Done Pro Monthly',
+      product: productId,
+      tiers_mode: null,
+      transform_usage: null,
+      trial_period_days: null,
+      usage_type: 'licensed',
+    },
+    price: {
+      id: planId,
+      object: 'price',
+      active: true,
+      billing_scheme: 'per_unit',
+      created: 1583259953,
+      currency: 'usd',
+      currency_options: {
+        usd: {
+          custom_unit_amount: null,
+          tax_behavior: 'exclusive',
+          unit_amount: 500,
+          unit_amount_decimal: '500',
+        },
+      },
+      custom_unit_amount: null,
+      livemode: false,
+      lookup_key: null,
+      metadata: {
+        productOrder: '1',
+      },
+      nickname: '123Done Pro Monthly',
+      product: productId,
+      recurring: {
+        aggregate_usage: null,
+        interval: 'month',
+        interval_count: 1,
+        trial_period_days: null,
+        usage_type: 'licensed',
+      },
+      tax_behavior: 'exclusive',
+      tiers_mode: null,
+      transform_quantity: null,
+      type: 'recurring',
+      unit_amount: 500,
+      unit_amount_decimal: '500',
```

Comment:
Repeated values so won't leave repeated comments, but similar applies to above items